### PR TITLE
Add multicollection support

### PR DESCRIPTION
**Pull request by alarbada**
### Description

Adds multicollection support for the sqs destination connector.

### TODO

- [x] Update docs
- [x] Add support for specifying queue name with a go template (as done with kinesis and others).

### Quick checks:

- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-sqs/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

**Review by raulb — COMMENTED** [review]
@alarbada is this being documented somewhere? (thinking about the user, if they'd know how to configure it or even be aware of the existence of a multicollection support)

**Comment by alarbada** [comment]
@raulb yeah sorry, I should've started the pull request as a draft instead. Now it should be complete!

File: README.md [hunk 1]
```diff
@@ -31,14 +33,14 @@ The configuration passed to `Configure` can contain the following fields:
...
+| `aws.secretAccessKey`   | AWS Secret Access Key                                                                                                                                                                                                                             | yes      | "SECRET_ACCESS_KEY"                          |
+| `aws.region`            | AWS SQS Region                                                                                                                                                                                                                                    | yes      | "us-east-1"                                  |
+| `aws.queue`             | AWS SQS Queue Name. It can contain a [Go template](https://pkg.go.dev/text/template) that will be executed for each record to determine the queue name. By default, the queue name will come from the `opencdc.collection` record metadata field. | yes      | `{{ index .Metadata "opencdc.collection" }}` |
```
Comment (raulb) [inline README.md:41]: This one shouldn't be required now, right?

File: destination/config.go [hunk 2]
```diff
@@ -21,6 +21,9 @@ import "github.com/conduitio-labs/conduit-connector-sqs/common"
 type Config struct {
 	common.Config
 
+	// AWSQueue is the sqs queue name
+	AWSQueue string `json:"aws.queue" validate:"required"`
```
Comment (raulb) [inline destination/config.go:25]: If we default to `opencdc.collection` this value shouldn't be required.

File: README.md [hunk 3]
```diff
@@ -14,6 +14,8 @@ The source connector pulls data from the Amazon SQS Queue. As messages come in,
 
 The destination connector batches incoming records into 10 and pushes them to the Amazon SQS Queue. Any fields defined in the metadata of the record will be passed as Message Attributes, and the json encoding of the record will be passed as the Message Body.
 
+It supports multicollection mode, where the sqs queue name is determined at runtime, based on the `opencdc.collection` metadata field or a given go template. If no queue name is found, the record will be written to the specified `aws.queue` parameter.
```
Comment (raulb) [inline README.md:17]: This would need to be rephrased once we change the default behavior.

File: connector_test.go [hunk 4]
```diff
@@ -95,6 +90,188 @@ func TestFifoQueues(t *testing.T) {
...
+		genRecord(defaultQueue.Name),
+		genRecord(defaultQueue.Name),
+
+		genRecord(testQueue1.Name),
+		genRecord(testQueue1.Name),
+
+		genRecord(testQueue2.Name),
+
+		genRecord(defaultQueue.Name),
+	}
```
Comment (raulb) [inline connector_test.go:147]: nit. I'd remove the empty lines here.

**Review by raulb — COMMENTED** [review]
@alarbada code looks good. I only have a comment on the default behavior, and about not requiring the queue name on destination. It is my understanding that, if we default to the metadata `opencdc.collection`, this value shouldn't be required. I would also love seeing a demo about this one working! Thank you for putting all the effort 🙏

File: destination/destination.go [hunk 5]
```diff
@@ -59,29 +63,47 @@ func (d *Destination) Configure(ctx context.Context, cfg config.Config) error {
...
+	d.queueURLMap = newQueueURLMap(d.svc)
 
-	// Get URL of queue
-	urlResult, err := d.svc.GetQueueUrl(ctx, queueInput)
-	if err != nil {
-		return fmt.Errorf("failed to get sqs queue url : %w", err)
-	}
+	if !isGoTemplate(d.config.QueueName) {
+		// we can be nice to the user and inform them that the queue that they did
+		// specify exists and that the connector has access to it.
```
Comment (maha-hajja) [inline destination/destination.go:94]: is this comment a todo? we can address this and delete the comment if so

Comment (alarbada) [reply destination/destination.go:94]: It isn't, but I agree that the wording isn't really clear. Let me fix that.

File: README.md [hunk 6]
```diff
@@ -31,14 +40,17 @@ The configuration passed to `Configure` can contain the following fields:
...
+| `aws.queue` (as a sqs queue name) | AWS queue name                                                                                                                                                           | no       | "QUEUE_NAME"                                 |
+| `aws.queue` (as a go template)    | Go template that is evaluated at runtime against a record to obtain the queue name                                                                                       | no       | `{{ index .Metadata "opencdc.collection" }}` |
+| `aws.queue` (empty)               | Makes the connector fetch the queue name from the `opencdc.collection`                                                                                                   | no       |                                              |
+| `aws.queue.useQueue`              | Forces the connector to write to the specified queue name, ignoring the `opencdc.collection` metadata field. Only relevant when `aws.queue` is given as a sqs queue name | no       |
```
Comment (raulb) [inline README.md:51]: By simplifying the usage and defaulting to `opencdc.collection` except when specified, we shouldn't have the need for this extra configuration parameter leaving the configuration much simpler.

File: destination/config.go [hunk 7]
```diff
@@ -21,6 +21,13 @@ import "github.com/conduitio-labs/conduit-connector-sqs/common"
 type Config struct {
 	common.Config
 
+	// QueueName is the sqs queue name
+	QueueName string `json:"aws.queue" default:"{{ index .Metadata \"opencdc.collection\" }}"`
+
+	// UseQueueName makes the connector always write to the specified queue name.
+	// Only relevant when `aws.queue` is given as a sqs queue name.
+	UseQueueName bool `json:"aws.queue.useQueue" default:"false"`
```
Comment (raulb) [inline destination/config.go:29]: This would go away if making the changes mentioned above.

**Review by raulb — COMMENTED** [review]
@alarbada code is looking good, but I would simplify the configuration options around the queue name on the destination and using it always as a go template removing the extra configuration field to solve ambiguity.

File: README.md [hunk 8]
```diff
@@ -10,10 +10,19 @@ Run `make build` to build the connector.
...
 The source connector pulls data from the Amazon SQS Queue. As messages come in, the source connector grabs a single message from the Amazon SQS Queue, formats it for the destination connector as a new record, and sends it. The Message Body of the SQS Message is formatted into a record's payload, and the Message Attributes are passed as record metadata. After the record has been acknowledged, the source connector deletes the message from the Amazon SQS Queue.
 
-## Destinaton
+## Destination
 
 The destination connector batches incoming records into 10 and pushes them to the Amazon SQS Queue. Any fields defined in the metadata of the record will be passed as Message Attributes, and the json encoding of the record will be passed as the Message Body.
 
+It can work in three modes, depending of the `aws.queue` parameter:
+
+- with a queue name: the connector tries to obtain the queue name from the `opencdc.collection`. If none found, defaults to the given `aws.queue`.
```
[the remark this answers is not in the document]
Comment (alarbada) [reply README.md:19]: Yeah, that makes sense, I might have overcomplicated it a little bit too much. I'll follow what kafka does.

**Comment by alarbada** [comment]
@maha-hajja @raulb latest commits apply the pr suggestions and remove repeated code in the multicollection tests, nothing more!

**Review by raulb — APPROVED** [review]
LGTM @alarbada . Great work

**Pull request merged**